File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.py
import heapq
from collections import Counter

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        cnt = Counter(nums)

        min_heap = []

        for num, freq in cnt.items():
            heapq.heappush(min_heap, (freq, num))

            if len(min_heap) > k:
                heapq.heappop(min_heap)

        res = []

        while min_heap:
            freq, num = heapq.heappop(min_heap)
            res.append(num)

        return res

File: Data Structures & Algorithms/is-anagram/submission-5.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()) return false;

        unordered_map<char, int> cnt;

        for(auto c : s) cnt[c]++;

        for(auto c : t){
            if(cnt[c] == 0){
                return false;
            }
            cnt[c] -= 1;
        }
        return true;
    }
};
